pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2272,6 +2282,8 @@ type SSSDSidecarAdditionalFile struct {
 }
 
 // NetworkSpec for Ceph includes backward compatibility code
+// +kubebuilder:validation:XValidation:rule="!has(self.provider) || (self.provider != 'multus' || (self.provider == 'multus' && size(self.selectors) > 0))",message="at least one network selector must be specified when using the multus"
+// +kubebuilder:validation:XValidation:rule="!has(self.provider) || (oldSelf.provider != 'host' || (oldSelf.provider == 'host' && self.provider != 'multus'))",message="invalid update: network provider from 'host' to 'multus' provider change is not allowed"
```

Comment:
The NetworkProvider type isn't actually nullable since it's a string. This is a bug IMO. I think we can omit `!has(self.provider)` from the spec if we remove the incorrect `// +nullable` from the `Provider NetworkProviderType` line.

```suggestion
// +kubebuilder:validation:XValidation:rule="self.provider != 'multus' || (self.provider == 'multus' && size(self.selectors) > 0)",message="at least one network selector must be specified when using the multus"
// +kubebuilder:validation:XValidation:rule="oldSelf.provider != 'host' || (oldSelf.provider == 'host' && self.provider != 'multus')",message="invalid update: network provider from 'host' to 'multus' provider change is not allowed"
```

Reply:
we need to add `!has(self.provider)` check since we are evaluating it later in the rule and with this it will through error `unknow key ...` also it is not related to `// +nullable`